plugins/outputs/sematext/sematext.go

```diff
@@ -0,0 +1,150 @@
+package sematext
+
+import (
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/outputs/sematext/processors"
+	"github.com/influxdata/telegraf/plugins/outputs/sematext/sender"
+	"github.com/influxdata/telegraf/plugins/outputs/sematext/serializer"
+	"net/http"
+	"net/url"
+)
+
+const (
+	defaultSematextMetricsReceiverUrl = "https://spm-receiver.sematext.com"
+)
+
+// Sematext struct contains configuration read from Telegraf config and a few runtime objects.
+// We'll use one separate instance of Telegraf for each monitored service. Therefore, token for particular service
+// will be configured on Sematext output level
+type Sematext struct {
+	ReceiverUrl string          `toml:"receiverUrl"`
+	Token       string          `toml:"token"`
+	ProxyServer string          `toml:"proxyServer"`
+	Username    string          `toml:"username"`
+	Password    string          `toml:"password"`
+	Log         telegraf.Logger `toml:"-"`
+
+	metricsUrl   string
+	sender       *sender.Sender
+	senderConfig *sender.Config
+	serializer   serializer.MetricSerializer
+	processors   []processors.Processor
+}
+
+// TODO add real sample config
+const sampleConfig = `
+  ## Sample config for Sematext output
+`
+
+// Connect is no-op for Sematext output plugin, everything was set up before in Init() method
+func (s *Sematext) Connect() error {
+	return nil
+}
+
+// Close Closes the Sematext output
+func (s *Sematext) Close() error {
+	s.sender.Close()
+	return nil
+}
+
+// SampleConfig Returns a sample configuration for the Sematext output
+func (s *Sematext) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the description for the Sematext output
+func (s *Sematext) Description() string {
+	return "Use telegraf to send metrics to Sematext"
+}
+
+// Init performs full initialization of Sematext output
+func (s *Sematext) Init() error {
+	if len(s.Token) == 0 {
+		s.Log.Error("'token' is a required field for Sematext output")
```

Comment:
I assume the error is logged in the caller context. If so, this line is redundant.

Reply:
That was my assumption too, but I saw some plugins were logging + returning. Looking again, it seems majority just returns, so I'll adjust.